ci_framework/roles/ci_network/tasks/nm-cleanup.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Fetch routes as a json
+  register: ip_routes
+  ansible.builtin.command:
+    cmd: ip -j ro ls
+
+- name: Get nm existing ethernet connections
+  register: nm_ethernet
+  ansible.builtin.shell:  # noqa risky-shell-pipe
+    cmd: nmcli -t -f TYPE,DEVICE,NAME con | grep ethernet
+
+- name: Prepare network data for proper consumption
+  ansible.builtin.set_fact:
```

Comment:
This task can be removed by adding the declared facts as variables of this one: https://github.com/openstack-k8s-operators/ci-framework/pull/586/files#diff-7bd038433c1242ed1f5b21ab2af6c230f41e76370a5ccde53ac3ef7435e9522cR23